esp32-wroom-rp/src/protocol.rs

```diff
@@ -392,7 +393,7 @@ pub(crate) trait ProtocolInterface {
     fn get_conn_status(&mut self) -> Result<ConnectionStatus, Error>;
     fn set_dns_config(&mut self, dns1: IpAddress, dns2: Option<IpAddress>) -> Result<(), Error>;
     fn req_host_by_name(&mut self, hostname: &str) -> Result<u8, Error>;
-    fn get_host_by_name(&mut self) -> Result<[u8; MAX_NINA_RESPONSE_LENGTH], Error>;
+    fn get_host_by_name(&mut self) -> Result<NinaResponseBuffer, Error>;
```

Comment:
Maybe we could have a couple of different NinaResponseBuffer types:
1. A shorter one, 255 bytes in length and maybe called `SmallNinaResponseBuffer`
2. A larger one intended for things like HTTP response documents and maybe called `LargeNinaResponseBuffer`